code/game/area/areas.dm

```diff
@@ -118,9 +123,10 @@
 
 	return INITIALIZE_HINT_LATELOAD
 
-/area/LateInitialize()
-	. = ..()
-	power_change()		// all machines set to current power level, also updates lighting icon
+/area/proc/create_powernet()
+	powernet = new()
+	powernet.powernet_area = src
+	return powernet
```

Comment:
I think this proc should contain the code from lines 94 to 105. because this *does* create the powernet but not entirely, half of the powernet is setup in the initialise proc.

What happens if this is called on an area with a powernet already? is there ever a need to replace a powernet? maybe if we ever have a powernet subtype?

Reply:
done